datadog/resource_datadog_role.go

```diff
@@ -123,9 +133,19 @@ func resourceDatadogRoleCustomizeDiff(ctx context.Context, diff *schema.Resource
 	for _, permI := range perms.List() {
 		perm := permI.(map[string]interface{})
 		permID := perm["id"].(string)
-		if _, ok := validPerms[permID]; !ok {
+
+		permAttributes, permissionExists := allPerms[permID]
+
+		if !permissionExists {
 			return fmt.Errorf(
-				"permission with ID %s is restricted and cannot be managed by terraform or does not exist, remove it from your configuration",
+				"permission with ID %s does not exist, remove it from your configuration",
+				permID,
+			)
+		}
+
+		if permAttributes.IsDefaultPermission && !defaultPermissionsOptOut.(bool) {
```

Comment:
Nit. Should this change be reverted to avoid new users from trying to use it

Reply:
good call! will keep this line of code for now and assess after code freeze. if we haven't started any of the rollout process yet, i'll remove it.